File: src/java/leetcode/editor/en/P429_NAryTreeLevelOrderTraversal.java
package java.leetcode.editor.en;
import java.util.*;
import java.util.function.*;
import java.leetcode.editor.en.definition.*;

public class P429_NAryTreeLevelOrderTraversal {
//leetcode submit region begin(Prohibit modification and deletion)
/*
// Definition for a Node.
class Node {
    public int val;
    public List<Node> children;

    public Node() {}

    public Node(int _val) {
        val = _val;
    }

    public Node(int _val, List<Node> _children) {
        val = _val;
        children = _children;
    }
};
*/

class Solution {
    public List<List<Integer>> levelOrder(Node root) {
        List<List<Integer>> ans = new ArrayList<>();
        helper(root, 0, ans);
        return ans;
    }
    private void helper(Node root, int level, List<List<Integer>> ans) {
        if (root == null) return;
        if (level == ans.size()) ans.add(new ArrayList<>());
        ans.get(level).add(root.val);
        for (Node child : root.children) {
            helper(child, level+1, ans);
        }
    }
}
//leetcode submit region end(Prohibit modification and deletion)
}

File: src/kt/leetcode/editor/en/P429_NAryTreeLevelOrderTraversal.kt
package kt.leetcode.editor.en;
import java.util.*;
import java.util.function.*;
import kt.leetcode.editor.en.definition.*;

public class P429_NAryTreeLevelOrderTraversal {
//leetcode submit region begin(Prohibit modification and deletion)
/**
 * Definition for a Node.
 * class Node(var `val`: Int) {
 *     var children: List<Node?> = listOf()
 * }
 */

class Solution {
    fun levelOrder(root: Node?): List<List<Int>> {
        val ans = ArrayList<ArrayList<Int>>()
        val queue = ArrayDeque<Node>()
        root?.let { queue.offer(it) }
        while (queue.isNotEmpty()) {
            var size = queue.size
            ans.add(ArrayList())
            while (size-- != 0) {
                val node = queue.poll()
                ans.last().add(node.`val`)
                node.children.forEach { queue.offer(it) }
            }
        }
        return ans
    }
}
//leetcode submit region end(Prohibit modification and deletion)
}
